### PR TITLE
WaitForARMFeedback: Add Runcommand resource type for VM

### PR DESCRIPTION
**Pull request by koralski**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by koralski** [comment]
Spec is posted here: https://microsoft.sharepoint.com/teams/ComputeVM/Shared%20Documents/Managed%20Services%20Team/Run%20Command%20V2%20Design.docx?web=1

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 1]
```diff
@@ -243,6 +243,295 @@
           "final-state-via": "location"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands/{runCommandName}": {
+      "put": {
+        "tags": [
+          "VirtualMachineVMRunCommands"
+        ],
+        "operationId": "VirtualMachineVMRunCommands_CreateOrUpdate",
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:252]: I suggest VirtualMachineRunCommands_CreateOrUpdate

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 2]
```diff
@@ -243,6 +243,295 @@
           "final-state-via": "location"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands/{runCommandName}": {
+      "put": {
+        "tags": [
+          "VirtualMachineVMRunCommands"
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:250]: just VirtualMachineRunCommands

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 3]
```diff
@@ -243,6 +243,295 @@
...
+          "Create or update a run command.": {
+            "$ref": "./examples/CreateOrUpdateRunCommand.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "VirtualMachineVMRunCommands"
+        ],
+        "operationId": "VirtualMachineVMRunCommands_Update",
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:317]: VirtualMachineRunCommands_Update

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 4]
```diff
@@ -243,6 +243,295 @@
...
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create or update a run command.": {
+            "$ref": "./examples/CreateOrUpdateRunCommand.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "VirtualMachineVMRunCommands"
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:315]: VirtualMachineRunCommands

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 5]
```diff
@@ -243,6 +243,295 @@
...
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Update a simple gallery.": {
+            "$ref": "./examples/UpdateRunCommand.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "VirtualMachineVMRunCommands"
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:374]: VirtualMachineRunCommands

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 6]
```diff
@@ -243,6 +243,295 @@
...
+        ],
+        "operationId": "VirtualMachineVMRunCommands_Delete",
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:376]: VirtualMachineRunCommands_Delete

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 7]
```diff
@@ -243,6 +243,295 @@
...
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Delete a gallery.": {
+            "$ref": "./examples/DeleteRunCommand.json"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "VirtualMachineVMRunCommands"
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:427]: VirtualMachineRunCommands

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 8]
```diff
@@ -243,6 +243,295 @@
...
+        ],
+        "operationId": "VirtualMachineVMRunCommands_Get",
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:429]: VirtualMachineRunCommands_Get

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 9]
```diff
@@ -243,6 +243,295 @@
...
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VirtualMachineRunCommand"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a gallery.": {
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:476]: is it for gallery?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 10]
```diff
@@ -243,6 +243,295 @@
...
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "No Content"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Delete a gallery.": {
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:420]: I think you mean delete a run command

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 11]
```diff
@@ -243,6 +243,295 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VirtualMachineRunCommand"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Update a simple gallery.": {
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:367]: run command

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 12]
```diff
@@ -243,6 +243,295 @@
...
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VirtualMachineVMRunCommandsListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List galleries in a resource group.": {
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:527]: list run commands

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 13]
```diff
@@ -243,6 +243,295 @@
...
+          "Get a gallery.": {
+            "$ref": "./examples/GetRunCommand.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands": {
+      "get": {
+        "tags": [
+          "VirtualMachineVMRunCommands"
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:485]: VirtualMachineRunCommands

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 14]
```diff
@@ -243,6 +243,295 @@
...
+        ],
+        "operationId": "VirtualMachineVMRunCommands_List",
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:487]: VirtualMachineRunCommands_List

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 15]
```diff
@@ -446,6 +735,160 @@
...
+          "$ref": "./compute.json#/definitions/Resource"
+        }
+      ],
+      "description": "Describes a Virtual Machine run command."
+    },
+    "VirtualMachineRunCommandUpdate": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineRunCommandProperties"
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:871]: You have the same VirtualMachineRunCommandProperties for both PUT and PATCH, which mean, if any mandatory property of PUT is also mandatory in PATCH.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json [hunk 16]
```diff
@@ -446,6 +735,160 @@
...
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state, which only appears in the response."
+        },
+        "instanceView": {
+          "readOnly": true,
+          "$ref": "#/definitions/VirtualMachineRunCommandInstanceView",
+          "description": "The virtual machine run command instance view."
+        }
+      },
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:850]: All properties are optional?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json (hunk 15 above)
Comment (koralski) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:871]: yes

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json (hunk 16 above)
Comment (koralski) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json:850]: Yes. All optional.

**Comment by koralski** [comment]
This PR is moved to private swagger since this API is not public yet:
https://github.com/Azure/azure-rest-api-specs-pr/pull/1246

**Pull request closed without merging**